edit/edit.js

```diff
@@ -1284,47 +1300,61 @@ function beautify(event) {
 
 onDOMready().then(init);
 
-function init() {
-  initCodeMirror();
+function createEmptyStyle() {
   const params = getParams();
-  if (!params.id) {
-    // match should be 2 - one for the whole thing, one for the parentheses
-    // This is an add
-    $('#heading').textContent = t('addStyleTitle');
-    const section = {code: ''};
-    for (const i in CssToProperty) {
-      if (params[i]) {
-        section[CssToProperty[i]] = [params[i]];
-      }
+  const style = {
+    id: null,
+    name: '',
+    enabled: true,
+    sections: [{code: ''}]
+  };
+  for (const i in CssToProperty) {
+    if (params[i]) {
+      style.sections[0][CssToProperty[i]] = [params[i]];
     }
-    addSection(null, section);
-    editors[0].setOption('lint', CodeMirror.defaults.lint);
-    editors[0].focus();
-    // default to enabled
-    $('#enabled').checked = true;
-    initHooks();
-    setCleanGlobal();
-    updateTitle();
-    return;
   }
-  // This is an edit
-  $('#heading').textContent = t('editStyleHeading');
-  getStylesSafe({id: params.id}).then(styles => {
-    let style = styles[0];
-    if (!style) {
-      style = {id: null, sections: []};
-      history.replaceState({}, document.title, location.pathname);
+  return style;
+}
+
+function windowLoaded() {
+  if (document.readyState !== 'loading') {
+    return Promise.resolve();
+  }
+  return new Promise(resolve => {
+    window.addEventListener('load', function self() {
+      window.removeEventListener('load', self);
+      resolve();
+    });
+  });
+}
+
+function init() {
+  initCodeMirror();
+  const params = getParams();
+  return Promise.resolve().then(() => {
```

Comment:
There's no need to return anything from `init`.  Why did you add the promise here? Remove it.